src/Components/Blazor/Blazor/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -88,10 +87,10 @@ private void CreateServiceProvider()
             var services = new ServiceCollection();
             services.AddSingleton(_BrowserHostBuilderContext);
             services.AddSingleton<IWebAssemblyHost, WebAssemblyHost>();
-            services.AddSingleton<IJSRuntime, MonoWebAssemblyJSRuntime>();
+            services.AddSingleton<IJSRuntime>(WebAssemblyJSRuntime.Current);
 
             services.AddSingleton<IUriHelper>(WebAssemblyUriHelper.Instance);
-            services.AddSingleton<HttpClient>(s =>
```

Comment:
Is there a way to just generally turn off this rule for cleanup? I don't like simplifying things like this (referring to DI) and many places in unit tests where we're testing delegates and type inference.

Reply:
Yeah, I was a bit hesitant about doing this. It doesn't improve readibility and is very easy to break